src/dns.zig

```diff
@@ -52,13 +52,53 @@ pub fn createQuery(allocator: mem.Allocator, address: []const u8, qtype: QType)
     return message;
 }
 
+pub const EDNS = struct {
+    bufsize: u16 = 1232,
+    do_dnssec: bool = true,
+    // In the future, EDNS options will come here
+    
+    pub fn format(self: *const EDNS, comptime fmt: []const u8, options: std.fmt.FormatOptions, writer: anytype) @TypeOf(writer).Error!void {
+        _ = fmt; // We don't use these two parameters
+        _ = options;
+        try writer.print("  EDNS {{\n", .{});
+        try writer.print("    Payload size: {d}\n", .{self.bufsize});
+        try writer.print("    Do DNSSEC: {any}\n", .{self.do_dnssec});
+        try writer.print("  }}\n", .{});
+    }
+};
+
+pub fn createEDNSQuery(allocator: mem.Allocator, address: []const u8, qtype: QType, edns: EDNS) !Message {
+    var result = try createQuery(allocator, address, qtype);
+    result.header.additional_record_count += 1;
+    const domain = try DomainName.from_string(allocator, ".");
+    var flags: i32 = 0; // EDNS version 0 (RFC 6891)
+    if (edns.do_dnssec) {
+        flags = 0x8000;
```

Comment:
> What do these flag values represent? It would be nice to have the flags represented by a packed struct to avoid unnamed constants.

0x8000 is just the DO (DNSSEC OK) bit set, everything else clear (which means EDNS version 0). But I agree (it's in my second commit) that a packed struct would be nicer.

Reply:
OK, next version will have:

```

pub const PackedEDNSTTL = packed struct(u32) {
    extendedRcode: u8 = 0,
    version: u8 = 0, // EDNS version 0 (RFC 6891)
    z1: u7 = 0, // Zig put fields in the unexpected order inside a byte.
    do: bool = true,
    z2: u8 = 0,
};

...

pub fn createEDNSQuery(allocator: mem.Allocator, address: []const u8, qtype: QType, edns: EDNS) !Message {
...
    var flags = PackedEDNSTTL{};
    flags.do = edns.do_dnssec;
```

No more 0x800. But I'm still struggling to produce a proper patch now that [read,write}IntBig disappeared and that non-mutated vars are forbidden. There are a lot of things to change.

Reply:
I must update the code to work with the latest Zig master release. Asking any PR to manage that migration on top of a new feature is a lot.

I have yet to have the time to work on this recently, but handling that migration is next on my list.

Reply:
I suspect most of my Zig projects need to be updated. Dealing with breaking changes every couple of months becomes overwhelming when you have a couple of larger projects you're working on in your spare time.